index.js

```diff
@@ -163,39 +174,50 @@ function fastifyStatic (fastify, opts, next) {
-        if (err) {
-          return next(err)
-        }
-        const indexDirs = new Set()
-        const indexes = typeof opts.index === 'undefined' ? ['index.html'] : [].concat(opts.index || [])
-        for (let file of files) {
-          file = file.replace(sendOptions.root.replace(/\\/g, '/'), '').replace(/^\//, '')
-          const route = encodeURI(prefix + file).replace(/\/\//g, '/')
-          fastify.get(route, routeOpts, function (req, reply) {
-            pumpSendToReply(req, reply, '/' + file)
-          })
 
-          if (indexes.includes(path.posix.basename(route))) {
-            indexDirs.add(path.posix.dirname(route))
+      function globber (rootPath) {
+        glob(path.join(rootPath, globPattern), { nodir: true }, function (err, files) {
+          if (err) {
+            return next(err)
           }
-        }
-        indexDirs.forEach(function (dirname) {
-          const pathname = dirname + (dirname.endsWith('/') ? '' : '/')
-          const file = '/' + pathname.replace(prefix, '')
+          const indexDirs = new Set()
+          const indexes = typeof opts.index === 'undefined' ? ['index.html'] : [].concat(opts.index || [])
+          for (let file of files) {
+            file = file.replace(rootPath.replace(/\\/g, '/'), '').replace(/^\//, '')
+            const route = encodeURI(prefix + file).replace(/\/\//g, '/')
+            fastify.get(route, routeOpts, function (req, reply) {
+              pumpSendToReply(req, reply, '/' + file, rootPath)
+            })
 
-          fastify.get(pathname, routeOpts, function (req, reply) {
-            pumpSendToReply(req, reply, file)
-          })
+            if (indexes.includes(path.posix.basename(route))) {
+              indexDirs.add(path.posix.dirname(route))
+            }
+          }
+          indexDirs.forEach(function (dirname) {
+            const pathname = dirname + (dirname.endsWith('/') ? '' : '/')
+            const file = '/' + pathname.replace(prefix, '')
 
-          if (opts.redirect === true) {
-            fastify.get(pathname.replace(/\/$/, ''), routeOpts, function (req, reply) {
-              pumpSendToReply(req, reply, file.replace(/\/$/, ''))
+            fastify.get(pathname, routeOpts, function (req, reply) {
+              pumpSendToReply(req, reply, file, rootPath)
             })
-          }
+
+            if (opts.redirect === true) {
+              fastify.get(pathname.replace(/\/$/, ''), routeOpts, function (req, reply) {
+                pumpSendToReply(req, reply, file.replace(/\/$/, ''), rootPath)
+              })
+            }
+          })
         })
-        next()
-      })
+      }
+
+      // TODO: this causes fastify to fail when running with multiple root paths.
+      if (Array.isArray(sendOptions.root)) {
+        for (let i = 0; i < sendOptions.root.length; i++) {
+          globber(sendOptions.root[i])
+        }
+      } else {
+        globber(sendOptions.root)
+      }
```

Comment:
This loop seems to cause issues when registering fastify routes for each of the matching glob paths. The first iteration is fine but the second fails.

Reply:
We now use `async.eachSeries`.